crates/matrix-sdk-crypto/src/gossiping/machine.rs

```diff
@@ -1431,11 +1451,13 @@ mod tests {
             .unwrap()
             .get(&DeviceIdOrAllDevices::DeviceId(alice_device_id().to_owned()))
             .unwrap();
-        let content: ToDeviceRoomEncryptedEventContent = content.deserialize_as().unwrap();
+        let content: ToDeviceEncryptedEventContent = content.deserialize_as().unwrap();
 
         bob_machine.mark_outgoing_request_as_sent(id).await.unwrap();
 
-        let event = ToDeviceEvent { sender: bob_id().to_owned(), content };
+        let event =
+            ToDeviceEvent { sender: bob_id().to_owned(), content, other: Default::default() };
+        let event = Raw::new(&event).unwrap().deserialize_as().unwrap();
```

Comment:
So we're converting something to JSON only to deserialize it again? If we really want to do that, I think we should move it to a helper function like

```rust
fn json_convert<T, U>(value: &T) -> serde_json::Result<U>
where
    T: Serialize,
    U: Deserialize,
{
    let json = serde_json::to_string(value)?;
    serde_json::from_str(&json)
}
```